pika/adapters/select_connection.py

```diff
@@ -310,12 +316,28 @@ def update_handler(self, fileno, events):
         :param int events: The event mask
 
         """
-        self.events = events
+
+        for fd_set, ev in zip(
+            [self.r_fds, self.w_fds, self.e_fds]
+            [READ, WRITE, ERROR]):
+            
+            if events & ev:
+                fd_set.add(fileno)
+            else:
+                fd_set.discard(fileno)
+
+    def remove_handler(self, fileno):
+        """Remove a fileno to the set
+        
+        :param int fileno: The file descriptor
+
+        """
+        del self.filenos[fileno]
```

Comment:
This is not enough. `fileno` might still be in `self.r_fds`, etc. Need something like this:
```
self.update_handler(fileno, 0)
del self.filenos[fileno]
```